kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -78,15 +79,14 @@ case class LambdaFunc(parameter: LambdaVar, body: LambdaTerm) extends LambdaTerm
 
   override val freeVars: Set[String] = body.freeVars - parameter.name
 
-  private[ast] def nestedFunctions(): (List[LambdaVar], LambdaTerm) = {
-    @tailrec
-    def go(func: LambdaFunc, acc: List[LambdaVar]): (List[LambdaVar], LambdaTerm) =
-      func match {
-        case LambdaFunc(v, f: LambdaFunc) => go(f, v :: acc)
-        case LambdaFunc(v, b)             => ((v :: acc).reverse, b)
-      }
-    go(this, Nil)
-  }
+  @hiddenargs
+  @tailrec
+  private[ast] def nestedFunctions(@hidden func: LambdaFunc     = this,
```

Comment:
I'm sorry but this is just straight up ugly. Why not get rid of that redundant argument and pattern match the body of the LambdaFunc?

```scala
def nestedFunctions(@hidden acc: List[LambdaVar] = Nil): (List[LambdaVar], LambdaTerm) =
  body match {
    case f: LambdaFunc => f.nestedFunctions(parameter :: acc)
    case _             => ((parameter :: acc).reverse, body)
  }
```